tracing-api/src/main/java/com/palantir/tracing/api/OpenSpan.java

```diff
@@ -71,9 +72,14 @@
     public static Builder builder() {
         return new Builder()
                 // TODO(rfink): Use direct access to system microseconds when moving to Java8 / Java9
-                .startTimeMicroSeconds(System.currentTimeMillis() * 1000)
+                .startTimeMicroSeconds(getNowInMicroSeconds())
                 .startClockNanoSeconds(System.nanoTime());
     }
 
+    private static long getNowInMicroSeconds() {
+        Instant now = Instant.now();
+        return (1000000 * now.getEpochSecond()) + (now.getNano() / 1000);
+    }
```

Comment:
We run this hundreds to thousands of times per request, regardless of whether our trace is sampled.
Does this have a measurable performance impact?

Reply:
I have not tested the performance impact. Do you have a suggested way of doing that?

Also, as a separate thing issue, should we stop calculating those timestamps for unsampled traces?

Reply:
Looks like a 20-25% regression in clock time on java 8, and 50% regression on java 12. I think that's a reasonable cost for more accurate values. If we see problems we can optimize by not calculating times for unsampled traces.